### PR TITLE
Ndcum 361

### PR DESCRIPTION
**Pull request by hsimonoko**
installed shellcheck and fixed .sh file errors

### CODE REVIEW

File: bamboo/build/write_inject_vars.sh [hunk 1]
```diff
@@ -6,19 +6,20 @@ set -e
 # REPO=${bamboo.REPO}
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+echo "$SCRIPT_DIR"
```
Comment (michaeljb) [inline bamboo/build/write_inject_vars.sh]: Do we need this line?

```suggestion
```

Comment (hsimonoko) [reply bamboo/build/write_inject_vars.sh]: this was my solution to the unused variable warning- I can disable it instead if that's better

File: publish.sh [hunk 2]
```diff
@@ -19,35 +19,36 @@ RELEASE_NAME=$4
...
+if [ -n "${RELEASE_NAME}" ]; then
     KEY="${KEY}-${RELEASE_NAME}"
 fi
 KEY="${KEY}.zip"
 
 echo "Publishing ${LAMBDA_ZIP} to s3://${BUCKET}/${KEY} and lambda ${CUMULUS_PREFIX}-${LAMBDA_NAME}"
 
 # upload to S3
-${AWS_CLI} s3 cp /project/$(basename ${LAMBDA_ZIP}) s3://${BUCKET}/${KEY}
+"${AWS_CLI}" s3 cp "${LAMBDA_ZIP}" s3://"${BUCKET}"/"${KEY}"
```
Comment (michaeljb) [inline publish.sh]: I notice that `/project` was dropped from the path here, was that intentional?

File: run_tests.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+echo "shellcheck"
+find . -type f -name "*.sh" -exec shellcheck {} \;
```
Comment (michaeljb) [inline run_tests.sh]:
```suggestion
find . -type f -name "*.sh" -print0 | xargs -0 shellcheck
```

File: publish.sh [hunk 4]
```diff
@@ -56,4 +57,4 @@ function_name="${CUMULUS_PREFIX}-${LAMBDA_NAME}"
          --function-name "${function_name}" \
          --s3-bucket "${BUCKET}" \
          --s3-key "${KEY}") ||
-    echo "lambda function '${function_name}' not found; lambda function must be created via terraform deployment"
+    (echo "lambda function '${function_name}' not found; lambda function must be created via terraform deployment" && exit 1)
```
Comment (michaeljb) [inline publish.sh]: Was the `&& exit 1` suggested by shellcheck, or did you add it to change how this was working?

My quick browser searches at https://gist.github.com/nicerobot/53cee11ee0abbdc997661e65b348f375 didn't lead me to a related issue page

File: publish.sh (hunk 2 above)
Comment (hsimonoko) [reply publish.sh]: hmm no, oops. can I make a commit with that on here directly?

File: bamboo/build/write_inject_vars.sh (hunk 1 above)
Comment (michaeljb) [reply bamboo/build/write_inject_vars.sh]: Ah, I see. Yeah I think I'd prefer disabling it, or deleting the `SCRIPT_DIR` line.

Some of the scripts use `SCRIPT_DIR` to be sure of where we are in relation to Bamboo's file system, but if this script isn't using it then I feel confident we can delete it.

File: publish.sh (hunk 4 above)
Comment (hsimonoko) [reply publish.sh]: I'll go back and look, I'm guessing shellcheck suggested it though

File: bamboo/build/write_inject_vars.sh [hunk 5]
```diff
@@ -6,19 +6,20 @@ set -e
 # REPO=${bamboo.REPO}
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```
Comment (michaeljb) [inline bamboo/build/write_inject_vars.sh:8]:
```suggestion
```

File: bamboo/build/write_inject_vars.sh (hunk 1 above)
Comment (michaeljb) [inline bamboo/build/write_inject_vars.sh]:
```suggestion
```

File: publish.sh (hunk 2 above)
Comment (michaeljb) [reply publish.sh]: Yeah, should be able to via the GitHub GUI or pushing code locally. Here, the suggestion block makes it easy:

```suggestion
"${AWS_CLI}" s3 cp /project/"${LAMBDA_ZIP}" s3://"${BUCKET}"/"${KEY}"
```

File: publish.sh (hunk 4 above)
Comment (michaeljb) [reply publish.sh]: I just realized this change is from a commit I made, in October...hmm....

Comment (michaeljb) [reply publish.sh]: OBE by rebase

File: publish.sh [hunk 6]
```diff
@@ -19,35 +19,36 @@ RELEASE_NAME=$4
...
+if [ -n "${RELEASE_NAME}" ]; then
     KEY="${KEY}-${RELEASE_NAME}"
 fi
 KEY="${KEY}.zip"
 
 echo "Publishing ${LAMBDA_ZIP} to s3://${BUCKET}/${KEY} and lambda ${CUMULUS_PREFIX}-${LAMBDA_NAME}"
 
 # upload to S3
-${AWS_CLI} s3 cp /project/$(basename ${LAMBDA_ZIP}) s3://${BUCKET}/${KEY}
+"${AWS_CLI}" s3 cp /project/$(basename "${LAMBDA_ZIP}") s3://"${BUCKET}"/"${KEY}"
```
Comment (hsimonoko) [inline publish.sh]:
```suggestion
"${AWS_CLI}" s3 cp /project/"$(basename "${LAMBDA_ZIP}")" s3://"${BUCKET}"/"${KEY}"
```

**Review by michaeljb — APPROVED** [review]

**Pull request merged**